optaplanner-examples/pom.xml

```diff
@@ -149,6 +149,12 @@
       <groupId>com.thoughtworks.xstream</groupId>
       <artifactId>xstream</artifactId>
     </dependency>
+    <!-- Comfortable assertions -->
+    <dependency>
+      <groupId>org.assertj</groupId>
+      <artifactId>assertj-core</artifactId>
```

Comment:
As we've discussed before: JUnit 5 migration before introducing AssertJ, so we can see what we're missing after that. I can fast track the JUnit 5 upgrade.

Mixing several technologies to do the same thing, is bad. It interferes with code unity and code consistency.

Reply:
I am afraid JUnit 5 introduces nothing new to assertions, apart from asserting that exception has been thrown using lambda. The fluent API which reads very well and nice, descriptive failure messages are not provided by JUnit. The way I look at JUnit vs. AssertJ is that JUnit provides test execution, AssertJ assertions. I agree with you that we should not mix them in a single purpose - for assertions, we should either completely switch to AssertJ or stick with JUnit.

Reply:
@rsynek You speak my mind :-)

Reply:
Just taking part with @rsynek and @yurloc :) In QE's main test repo we have [a checkstyle rule](https://github.com/jboss-integration/bxms-qe-tests/blob/master/pom.xml#L803-L807) that makes using `import org.junit.Assert.*` forcing people to use assertj consistently.

Reply:
A PR cannot introduce a new technology alternative for *some* files, leaving the rest of the codebase in the old way. That's how you end up with XLS decision tables that use POI and XLS DRL templates that use jexcelAPI (and this was still fixable in reasonable time, the MVC vs MVP vs pre-MVP GWT forms in the workbench is another matter).
Furthermore, a PR should not mix 2 non-trivial things such as introducing multithreaded solving tests and introducing assertj in every JUnit test.

Do or do not. There is not try...ing to add assertj technology in some files as part of another issue. Let's revert this PR to JUnit style and merge this. Let's then upgrade to JUnit 5. And let's then do the AssertJ discussion, based on experience and data. Again, I am open to fast-track the JUnit 5 work to move forward on this.

Reply:
I agree with the point regarding consistency. I will change this PR to use plain jUnit assertions and do the big bang migration to AssertJ later.

Reply:
@ge0ffrey just for the record, there is already AssertJ being used in optaplanner-core. Take a look at DefaultPartitionedSearchPhaseTest.